File: C++/search/linear_search.cpp
#include<bits/stdc++.h>
using namespace std;

int search(vector<int> v, int find)
{
	int l= v.size();
	for(int i=0; i<l ; i++)
	{
		if(find==v[i])
			return i;
	}
	//if not found in the array
	return -1;
}

int main()
{
	vector<int> v;
	//array for example is: 1 3 5 2 4 6 7 9 8
	v.push_back(1);v.push_back(3);v.push_back(5);
	v.push_back(2);v.push_back(4);v.push_back(6);
	v.push_back(7);v.push_back(9);v.push_back(8);
	
	int n;
	//to search this element
	cin>>n;
	
	int l= search(v,n);
	if(l==-1)
		cout<<"Not found\n";
	else
		cout<<n<<" is found at position "<<l<<endl;
		
	return 0;
}

File: Go/LinearSearch.go
/*Copyright 2020 Mahima Goyal

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0
    
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

// Code for Linear Search in Go

package main

import (
	"fmt"
)

func main() {
	var arr = []int{1, 5, 2, -5, 8, 4, 9, -12}
	fmt.Printf("Array: %v\n", arr)
	var find int
	fmt.Printf("Please enter element to be found: ")
	fmt.Scan(&find)
	result := search(arr, find)
	fmt.Printf("Result: %d", result)
}

func search(list []int, item int) int {
	for i := range list {
		if list[i] == item {
			return i
		}
	}
	return -1
}
